broadinstitute/single_cell_portal_core
# Plotting sorted by expression (SCP-3779)

This sorts the trace we input to plotly in ascending order of gene expression, so that expressed values will be plotted on top of non-expressed values.  This adds a new synthetic study 'chicken_overplot' to test the behavior.   This adds about 1 second to plot time per 1MM cells plotted in localhost testing

BEFORE (note the 'A' is grey)
![image](https://user-images.githubusercontent.com/2800795/139729574-35d4dc16-8c6a-4e02-9079-b1e645ce3e0e.png)

AFTER (note the 'A' is now expressed)
![image](https://user-images.githubusercontent.com/2800795/139729649-699fbe0b-b394-4f63-9a5b-966ef75c7cdd.png)

This also cleans out an unused file from the chicken_raw_counts study.

TO TEST:
1.  in console, run `SyntheticStudyPopulator.populate('chicken_overplot')`
2. view the study,  and confirm the biosample overplotting -- hiding the red trace 'A' should reveal the purple 'A' behind it
3. search for gene Foo, and confirm that the 'A' shows an expression gradient.
4. comment out line 286 in `ScatterPlot.js` (the `expressionsWithIndices.sort...` line), and confirm the 'A' would be greyed out without the sorting
